jni/src/com_amazon_opendistroforelasticsearch_knn_index_v2011_KNNIndex.cpp

```diff
@@ -189,6 +209,13 @@ JNIEXPORT jlong JNICALL Java_com_amazon_opendistroforelasticsearch_knn_index_v20
         string spaceTypeString(spaceTypeCStr);
         env->ReleaseStringUTFChars(spaceType, spaceTypeCStr);
         has_exception_in_stack(env);
+
+        if(mapSpace.find(spaceTypeString) == mapSpace.end()) {
+            throw std::runtime_error("Space not found");
+        }
+
+        spaceTypeString = mapSpace[spaceTypeString];
+
```

Comment:
Minor: We can move this logic to a function and reuse?

Reply:
Yes it could